### PR TITLE
redesigning layout of unit test plugin

### PR DESCRIPTION
**Pull request by LianaHus**
fixing https://github.com/ethereum/remix-ide/issues/2711

### CODE REVIEW

File: src/app/tabs/test-tab.js [hunk 1]
```diff
@@ -239,12 +245,18 @@ module.exports = class TestTab extends ViewPlugin {
...
+      </div>
+    `
     const isSolidityActive = this.appManager.actives.includes('solidity')
     if (!currentFile || !isSolidityActive) {
       el.setAttribute('disabled', 'disabled')
-      if (!currentFile) el.setAttribute('title', 'No file selected')
-      if (!isSolidityActive) el.setAttribute('title', 'The solidity module should be activated')
+      if (!currentFile) {
+        el.setAttribute('title', 'No file selected')
+      } else if (!isSolidityActive) el.setAttribute('title', 'The solidity module should be activated')
```
Comment (Aniket-Engg) [inline src/app/tabs/test-tab.js:259]: Better message can be: `'Solidity Compiler' plugin should be activated`

File: src/app/tabs/test-tab.js [hunk 2]
```diff
@@ -255,50 +267,70 @@ module.exports = class TestTab extends ViewPlugin {
...
+          checked="true"
+        >
+        <label class="text-nowrap pl-2 mb-0" for="checkAllTests"> Select all </label>
+      </div>
+    `
+  }
+
+  infoButton () {
+    return yo`
+      <a class="btn border d-flex w-50 ml-2" title="Check out documentation." target="__blank" href="https://remix-ide.readthedocs.io/en/latest/unittesting.html">
```
Comment (Aniket-Engg) [inline src/app/tabs/test-tab.js:305]: `How to use` button currently shows an underline on text. I think that should be removed. 

Also, better link to redirect can be https://remix-ide.readthedocs.io/en/latest/unittesting.html#generate-test-file as plugin is already activated.

File: src/app/tabs/test-tab.js (hunk 1 above)
Comment (LianaHus) [reply src/app/tabs/test-tab.js:259]: it is like that in my local changes. I'll push them

File: src/app/tabs/test-tab.js [hunk 3]
```diff
@@ -239,12 +257,19 @@ module.exports = class TestTab extends ViewPlugin {
...
+      </button>
+    `
     const isSolidityActive = this.appManager.actives.includes('solidity')
     if (!currentFile || !isSolidityActive) {
       el.setAttribute('disabled', 'disabled')
-      if (!currentFile) el.setAttribute('title', 'No file selected')
-      if (!isSolidityActive) el.setAttribute('title', 'The solidity module should be activated')
+      if (!currentFile) {
+        el.setAttribute('title', 'No file selected')
+      } else if (!isSolidityActive) el.setAttribute('title', 'The "Solidity Plugin" should be activated')
```
Comment (Aniket-Engg) [inline src/app/tabs/test-tab.js:271]: title can better updated to `'Solidity Compiler' plugin should be activated`

File: src/app/tabs/test-tab.js [hunk 4]
```diff
@@ -255,50 +280,75 @@ module.exports = class TestTab extends ViewPlugin {
...
+          checked="true"
+        >
+        <label class="text-nowrap pl-2 mb-0" for="checkAllTests"> Select all </label>
+      </div>
+    `
+  }
+
+  infoButton () {
+    return yo`
+      <a class="btn border text-decoration-none pr-0 d-flex w-50 ml-2" title="Check out documentation." target="__blank" href="https://remix-ide.readthedocs.io/en/latest/unittesting.html">
```
Comment (Aniket-Engg) [inline src/app/tabs/test-tab.js:323]: better link to redirect can be https://remix-ide.readthedocs.io/en/latest/unittesting.html#generate-test-file

Comment (LianaHus) [reply src/app/tabs/test-tab.js:324]: will do

**Comment by ryestew** [comment]
The icons on the buttons are not changing color when the theme goes to the light theme to the dark theme - but the text is changing.  A number of times I thought that the stop icon was an unchecked checkbox.  So maybe the icon on stop is not necessary.  With play it adds some focus.

Also I don't think the generate and the how to use are worth being buttons.  They could both be links in the paragraph above.
Also in the paragraph above - it could say that there needs to be a file compiled and a test file to have the run button enabled.

**Comment by ryestew** [comment]
The link from the button How to use... doesn't provide much info - it is just the section about the generate a sample file.  So I think it would be better to link to the top of the docs - https://remix-ide.readthedocs.io/en/latest/unittesting.html
While it is a duplication of the link at the top of the page - it will get them to read about how to make unit tests.

**Comment by Aniket-Engg** [comment]
@ryestew top of the docs starts with plugin activation. As `How to use...` will appear only after plugin is activated, so IMHO it is fine to avoid that plugin activation section and link after it.

**Pull request merged**